socketmaster.go

```diff
@@ -30,9 +30,12 @@ func handleSignals(processGroup *ProcessGroup, c <-chan os.Signal, startTime int
 					time.Sleep(time.Duration(startTime) * time.Millisecond)
 				}
 
-				// A possible improvement woud be to only swap the
-				// process if the new child is still alive.
-				processGroup.SignalAll(syscall.SIGTERM, process)
+				if len(processGroup.set.set) > 1 {
+					processGroup.SignalAll(syscall.SIGTERM, process)
+				} else {
+					processGroup.SignalAll(syscall.SIGCHLD, nil)
```

Comment:
The issue is that SIGCHLD is something the process should only receive if one of it's own sub-processes dies. I think the idea was for the process to call `wait()` and get the exit status. If the process does that in this context it will be stuck on the `wait()` since none of it's sub-processes has died. Or that's my understanding: https://en.wikipedia.org/wiki/Unix_signal

If you want better cooperation during deploy I think socketmaster would require a control socket. But in that case have a look at [crank](https://github.com/pusher/crank) (which I also wrote). It's a bit more complicated but also more featureful.